testParseProcessDefinitionXXE breaks build

**Describe the bug**

I'm experiencing a failure running a normal Maven build on current `main`(#9cc4fd71)

```
[ERROR] Failures: 
[ERROR]   BpmnParseTest.shouldPreventXxeProcessing:1212 
Expecting throwable message:
  "ENGINE-09003 Could not parse 'org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseProcessDefinitionXXE.bpmn20.xml'. DOCTYPE ist nicht zulässig, wenn das Feature "http://apache.org/xml/features/disallow-doctype-decl" auf "true" gesetzt ist."
to contain:
  "DOCTYPE is disallowed when the feature "http://apache.org/xml/features/disallow-doctype-decl" set to true."
but did not.

Throwable that failed the check:

org.operaton.bpm.engine.ProcessEngineException: ENGINE-09003 Could not parse 'org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseProcessDefinitionXXE.bpmn20.xml'. DOCTYPE ist nicht zulässig, wenn das Feature "http://apache.org/xml/features/disallow-doctype-decl" auf "true" gesetzt ist.
        at org.operaton.bpm.engine.impl.util.EngineUtilLogger.parsingFailureException(EngineUtilLogger.java:54)
        at org.operaton.bpm.engine.impl.util.xml.Parse.execute(Parse.java:150)
        at org.operaton.bpm.engine.impl.bpmn.parser.BpmnParse.execute(BpmnParse.java:334)
        at org.operaton.bpm.engine.impl.bpmn.deployer.BpmnDeployer.transformDefinitions(BpmnDeployer.java:106)
        at org.operaton.bpm.engine.impl.AbstractDefinitionDeployer.transformResource(AbstractDefinitionDeployer.java:99)
        at org.operaton.bpm.engine.impl.AbstractDefinitionDeployer.parseDefinitionResources(AbstractDefinitionDeployer.java:74)
        at org.operaton.bpm.engine.impl.AbstractDefinitionDeployer.deploy(AbstractDefinitionDeployer.java:64)
        at org.operaton.bpm.engine.impl.persistence.deploy.cache.CacheDeployer$1.call(CacheDeployer.java:54)
        at org.operaton.bpm.engine.impl.persistence.deploy.cache.CacheDeployer$1.call(CacheDeployer.java:51)
        at org.operaton.bpm.engine.impl.interceptor.CommandContext.runWithoutAuthorization(CommandContext.java:554)
        at org.operaton.bpm.engine.impl.interceptor.CommandContext.runWithoutAuthorization(CommandContext.java:542)
        at org.operaton.bpm.engine.impl.persistence.deploy.cache.CacheDeployer.deploy(CacheDeployer.java:51)
        at org.operaton.bpm.engine.impl.persistence.deploy.cache.DeploymentCache.deploy(DeploymentCache.java:71)
        at org.operaton.bpm.engine.impl.persistence.entity.DeploymentManager.insertDeployment(DeploymentManager.java:64)
        at org.operaton.bpm.engine.impl.cmd.DeployCmd.deploy(DeployCmd.java:258)
        at org.operaton.bpm.engine.impl.cmd.DeployCmd.lambda$doExecute$0(DeployCmd.java:155)
        at org.operaton.bpm.engine.impl.interceptor.CommandContext.runWithoutAuthorization(CommandContext.java:554)
        at org.operaton.bpm.engine.impl.interceptor.CommandContext.runWithoutAuthorization(CommandContext.java:542)
        at org.operaton.bpm.engine.impl.cmd.DeployCmd.doExecute(DeployCmd.java:141)
        at org.operaton.bpm.engine.impl.cmd.DeployCmd.execute(DeployCmd.java:101)
        at org.operaton.bpm.engine.impl.cmd.DeployCmd.execute(DeployCmd.java:81)
        at org.operaton.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.operaton.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:111)
        at org.operaton.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.operaton.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:34)
        at org.operaton.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.operaton.bpm.engine.impl.interceptor.ExceptionCodeInterceptor.execute(ExceptionCodeInterceptor.java:55)
        at org.operaton.bpm.engine.impl.RepositoryServiceImpl.deployWithResult(RepositoryServiceImpl.java:104)
        at org.operaton.bpm.engine.impl.repository.DeploymentBuilderImpl.deployWithResult(DeploymentBuilderImpl.java:270)
        at org.operaton.bpm.engine.impl.repository.DeploymentBuilderImpl.deploy(DeploymentBuilderImpl.java:266)
        at org.operaton.bpm.engine.test.bpmn.parse.BpmnParseTest.lambda$shouldPreventXxeProcessing$0(BpmnParseTest.java:1210)
        at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:63)
        at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:904)
        at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1470)
        at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1313)
        at org.operaton.bpm.engine.test.bpmn.parse.BpmnParseTest.shouldPreventXxeProcessing(BpmnParseTest.java:1207)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
        at org.operaton.bpm.engine.test.ProcessEngineRule$1.evaluate(ProcessEngineRule.java:191)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: org.xml.sax.SAXParseException; lineNumber: 2; columnNumber: 10; DOCTYPE ist nicht zulässig, wenn das Feature "http://apache.org/xml/features/disallow-doctype-decl" auf "true" gesetzt ist.
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(AbstractSAXParser.java:1251)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(SAXParserImpl.java:637)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl.parse(SAXParserImpl.java:326)
        at java.xml/javax.xml.parsers.SAXParser.parse(SAXParser.java:197)
        at org.operaton.bpm.engine.impl.util.xml.Parse.execute(Parse.java:148)
        ... 68 more

[ERROR]   BpmnParseTest.testInvalidSubProcessWithConditionalStartEvent:163 expected presence of [The content of element 'bpmn2:conditionalEventDefinition' is not complete.], but was [ENGINE-09005 Could not parse BPMN process. Errors: 
* cvc-complex-type.2.4.b: Content des Elements 'bpmn2:conditionalEventDefinition' ist nicht vollständig. '{"http://www.omg.org/spec/BPMN/20100524/MODEL":documentation, "http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements, "http://www.omg.org/spec/BPMN/20100524/MODEL":condition}' wird erwartet. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testInvalidSubProcessWithConditionalStartEvent.bpmn | line 24 | column 78
* conditionalEventDefinition is not allowed on start event within a subprocess | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testInvalidSubProcessWithConditionalStartEvent.bpmn | line 24 | column 78]
[ERROR]   BpmnParseTest.testInvalidTransactionWithConditionalStartEvent:263 expected presence of [The content of element 'bpmn2:conditionalEventDefinition' is not complete.], but was [ENGINE-09005 Could not parse BPMN process. Errors: 
* cvc-complex-type.2.4.b: Content des Elements 'bpmn2:conditionalEventDefinition' ist nicht vollständig. '{"http://www.omg.org/spec/BPMN/20100524/MODEL":documentation, "http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements, "http://www.omg.org/spec/BPMN/20100524/MODEL":condition}' wird erwartet. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testInvalidTransactionWithConditionalStartEvent.bpmn | line 15 | column 78
* conditionalEventDefinition is not allowed on start event within a subprocess | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testInvalidTransactionWithConditionalStartEvent.bpmn | line 15 | column 78]
[ERROR]   BpmnParseTest.testParseInvalidConditionalBoundaryEvent:722->parseInvalidConditionalEvent:711 expected presence of [The content of element 'bpmn:conditionalEventDefinition' is not complete.], but was [ENGINE-09005 Could not parse BPMN process. Errors: 
* cvc-complex-type.2.4.b: Content des Elements 'bpmn:conditionalEventDefinition' ist nicht vollständig. '{"http://www.omg.org/spec/BPMN/20100524/MODEL":documentation, "http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements, "http://www.omg.org/spec/BPMN/20100524/MODEL":condition}' wird erwartet. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidConditionalBoundaryEvent.bpmn | line 22 | column 41
* Conditional event must contain an expression for evaluation. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidConditionalBoundaryEvent.bpmn | line 22 | column 41]
[ERROR]   BpmnParseTest.testParseInvalidEventSubprocessConditionalStartEvent:763->parseInvalidConditionalEvent:711 expected presence of [The content of element 'bpmn:conditionalEventDefinition' is not complete.], but was [ENGINE-09005 Could not parse BPMN process. Errors: 
* cvc-complex-type.2.4.b: Content des Elements 'bpmn:conditionalEventDefinition' ist nicht vollständig. '{"http://www.omg.org/spec/BPMN/20100524/MODEL":documentation, "http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements, "http://www.omg.org/spec/BPMN/20100524/MODEL":condition}' wird erwartet. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidEventSubprocessConditionalStartEvent.bpmn | line 19 | column 43
* Conditional event must contain an expression for evaluation. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidEventSubprocessConditionalStartEvent.bpmn | line 19 | column 43]
[ERROR]   BpmnParseTest.testParseInvalidIntermediateConditionalEvent:749->parseInvalidConditionalEvent:711 expected presence of [The content of element 'bpmn:conditionalEventDefinition' is not complete.], but was [ENGINE-09005 Could not parse BPMN process. Errors: 
* cvc-complex-type.2.4.b: Content des Elements 'bpmn:conditionalEventDefinition' ist nicht vollständig. '{"http://www.omg.org/spec/BPMN/20100524/MODEL":documentation, "http://www.omg.org/spec/BPMN/20100524/MODEL":extensionElements, "http://www.omg.org/spec/BPMN/20100524/MODEL":condition}' wird erwartet. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidIntermediateConditionalEvent.bpmn | line 11 | column 41
* Conditional event must contain an expression for evaluation. | resource org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testParseInvalidIntermediateConditionalEvent.bpmn | line 11 | column 41]
[ERROR]   BpmnParseTest.testXmlParsingErrors:360 expected presence of [The end-tag for element type "bpmndi:BPMNLabel" must end with a '>' delimiter], but was [ENGINE-09003 Could not parse 'org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.testXMLParsingErrors.bpmn'. Endtag für Elementtyp "bpmndi:BPMNLabel" muss mit einem ">"-Begrenzungszeichen enden.]
[INFO] 
[ERROR] Tests run: 15757, Failures: 7, Errors: 0, Skipped: 73
```

**To Reproduce**
Steps to reproduce the behavior:

Run on `main` branch:
```./mvnw clean verify```

--- a/engine/src/test/java/org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.java
+++ b/engine/src/test/java/org/operaton/bpm/engine/test/bpmn/parse/BpmnParseTest.java
@@ -16,33 +16,10 @@
  */
 package org.operaton.bpm.engine.test.bpmn.parse;
 
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
-import static org.hamcrest.CoreMatchers.containsString;
-import static org.hamcrest.CoreMatchers.not;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-
-import java.util.List;
-
-import org.operaton.bpm.engine.ActivityTypes;
-import org.operaton.bpm.engine.ParseException;
-import org.operaton.bpm.engine.Problem;
-import org.operaton.bpm.engine.ProcessEngineException;
-import org.operaton.bpm.engine.RepositoryService;
-import org.operaton.bpm.engine.RuntimeService;
-import org.operaton.bpm.engine.impl.bpmn.behavior.BoundaryConditionalEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.BoundaryEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.CompensationEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.EventSubProcessStartConditionalEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.EventSubProcessStartEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.IntermediateConditionalEventBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.NoneStartEventActivityBehavior;
-import org.operaton.bpm.engine.impl.bpmn.behavior.ThrowEscalationEventActivityBehavior;
+import org.junit.*;
+import org.junit.rules.RuleChain;
+import org.operaton.bpm.engine.*;
+import org.operaton.bpm.engine.impl.bpmn.behavior.*;
 import org.operaton.bpm.engine.impl.bpmn.helper.BpmnProperties;
 import org.operaton.bpm.engine.impl.bpmn.parser.BpmnParse;
 import org.operaton.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
@@ -69,12 +46,15 @@ import org.operaton.bpm.model.bpmn.Bpmn;
 import org.operaton.bpm.model.bpmn.BpmnModelInstance;
 import org.operaton.commons.testing.ProcessEngineLoggingRule;
 import org.operaton.commons.testing.WatchLogger;
-import org.junit.After;
-import org.junit.Assume;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.RuleChain;
+
+import java.util.List;
+import java.util.Locale;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.*;
 
 /**
  *
@@ -98,16 +78,21 @@ public class BpmnParseTest {
   public RuntimeService runtimeService;
   public ProcessEngineConfigurationImpl processEngineConfiguration;
 
+  private Locale defaultLocale;
+
   @Before
   public void setup() {
     repositoryService = engineRule.getRepositoryService();
     runtimeService = engineRule.getRuntimeService();
     processEngineConfiguration = engineRule.getProcessEngineConfiguration();
     processEngineConfiguration.setEnableXxeProcessing(false);
+    defaultLocale = Locale.getDefault();
+    Locale.setDefault(Locale.US);
   }
 
   @After
   public void tearDown() {
+    Locale.setDefault(defaultLocale);
     for (org.operaton.bpm.engine.repository.Deployment deployment : repositoryService.createDeploymentQuery().list()) {
       repositoryService.deleteDeployment(deployment.getId(), true);
     }

